cmd/snap-recovery-chooser/main_test.go

```diff
@@ -0,0 +1,466 @@
+	expect map[string]interface{}
+}
+
+func (s *mockedClientCmdSuite) mockSuccessfulResponse(c *C, rspSystems *main.ChooserSystems, reqSystemLabel *mockSystemRequestResponse) {
+	n := 0
+	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
+		switch n {
+		case 0:
+			c.Check(r.URL.Path, Equals, "/v2/systems")
+			err := json.NewEncoder(w).Encode(apiResponse{
+				Type:       "sync",
+				Result:     rspSystems,
+				StatusCode: 200,
+			})
+			c.Assert(err, IsNil)
+		case 1:
+			if reqSystemLabel == nil {
+				c.Fatalf("unexpected request to %q", r.URL.Path)
+			}
+			c.Check(r.URL.Path, Equals, "/v2/systems/"+reqSystemLabel.label)
+			c.Check(r.Method, Equals, "POST")
+
+			var data map[string]interface{}
+			err := json.NewDecoder(r.Body).Decode(&data)
+			c.Assert(err, IsNil)
+			c.Check(data, DeepEquals, reqSystemLabel.expect)
+
+			rspType := "sync"
+			var rspData map[string]string
+			if reqSystemLabel.code >= 400 {
+				rspType = "error"
+				rspData = map[string]string{"message": "failed in mock"}
+			}
+			err = json.NewEncoder(w).Encode(apiResponse{
+				Type:       rspType,
+				Result:     rspData,
+				StatusCode: reqSystemLabel.code,
+			})
+			c.Assert(err, IsNil)
+		default:
+			c.Fatalf("expected to get 1 requests, now on %d", n+1)
+		}
+		n++
+	})
+}
+
+type apiResponse struct {
+	Type       string      `json:"type"`
+	Result     interface{} `json:"result"`
+	StatusCode int         `json:"status-code"`
+}
+
+func (s *mockedClientCmdSuite) TestMainChooserWithTool(c *C) {
+	r := main.MockDefaultMarkerFile(s.markerFile)
+	defer r()
+	// sanity
+	c.Assert(s.markerFile, testutil.FilePresent)
+
+	mockCmd := testutil.MockCommand(c, "tool", `
+echo '{"label":"label","action":{"mode":"install","title":"reinstall"}}'
+`)
+	defer mockCmd.Restore()
+	r = main.MockChooserTool(func() (*exec.Cmd, error) {
+		return exec.Command(mockCmd.Exe()), nil
+	})
+	defer r()
+
+	s.mockSuccessfulResponse(c, mockSystems, &mockSystemRequestResponse{
+		code:  200,
+		label: "label",
+		expect: map[string]interface{}{
+			"action": "do",
+			"mode":   "install",
+			"title":  "reinstall",
+		},
+	})
+
+	err := main.Chooser(client.New(&s.config))
+	c.Assert(err, IsNil)
+	c.Assert(mockCmd.Calls(), DeepEquals, [][]string{
+		{"tool"},
+	})
+
+	c.Assert(s.markerFile, testutil.FileAbsent)
+}
+
+func (s *mockedClientCmdSuite) TestMainChooserToolNotFound(c *C) {
+	r := main.MockDefaultMarkerFile(s.markerFile)
+	defer r()
+	// sanity
+	c.Assert(s.markerFile, testutil.FilePresent)
+
+	s.mockSuccessfulResponse(c, mockSystems, nil)
+
+	r = main.MockChooserTool(func() (*exec.Cmd, error) {
+		return nil, fmt.Errorf("tool not found")
+	})
+	defer r()
+
+	err := main.Chooser(client.New(&s.config))
+	c.Assert(err, ErrorMatches, "cannot locate the chooser UI tool: tool not found")
+
+	c.Assert(s.markerFile, testutil.FileAbsent)
+}
+
+func (s *mockedClientCmdSuite) TestMainChooserStdout(c *C) {
+	os.Setenv("SNAPPY_TESTING_USE_STDOUT", "1")
```

Comment:
This makes me wonder, should the side that reads this be using `snapdenv`?

Reply:
The flag is only used in the snap-recovery-chooser binary.

Reply:
if it's very specific to testing this code I would say no. For the moment the main candidates for snapdenv are things used by more than one package, or well advertised to users